core/src/main/java/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java

```diff
@@ -173,12 +174,33 @@ private Answer execute(final SetupKeyStoreCommand cmd) {
         return new SetupKeystoreAnswer(result.getDetails());
     }
 
+    private void scpFileToIdentifyPatching(String routerIp, String path, String filename, String content) throws InterruptedException {
```

Comment:
nit - maybe rename to a general utility method name - basically you're trying to copy and not giving up on the first fail?

Reply:
The idea of this is actually to see if the path `/usr/local/cloud/systemvm/conf` is created i.e., the patching process has actually completed. As in the earlier workflow, we wouldn't come to this point without ssh being made available - and ssh service would be enabled as part of postinit. But in the new workflow, ssh service is enabled right at the beginning of the patching process, hence, we'd need to wait for some time before the directory is created during the bootstrap process